src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -886,28 +885,14 @@ private static boolean isAstInside(DetailAST tree, DetailAST ast) {
      * @param ident ident token.
      * @return true if field can be referenced from a static context.
      */
-    private boolean canBeReferencedFromStaticContext(DetailAST ident) {
-        AbstractFrame variableDeclarationFrame = findFrame(ident, false);
-        while (variableDeclarationFrame.getType() == FrameType.BLOCK_FRAME
-            || variableDeclarationFrame.getType() == FrameType.FOR_FRAME) {
```

Comment:
This mutation had a cascading effect, and so the end result is the frame of the `ident` does not matter.

According to the callers, the frame can only be `CLASS_FRAME`, `METHOD_FRAME`, `BLOCK_FRAME`.

This means the frame can never be `FOR_FRAME`, so that is why it can be removed without issue.